usaspending_api/broker/management/commands/fix_locations.py

```diff
@@ -0,0 +1,192 @@
+                    curs.execute(qry)
+                    elapsed = time.time() - start
+                    logger.info('{}: ID {} to {}, {} s'.format(descrip, floor, ceiling, elapsed))
+
+    def find_batches(self, curs, table, options):
+        batch = options['batch']
+        curs.execute(self.BOUNDARY_FINDER.format(table))
+        (lowest, highest) = curs.fetchone()
+        floor = (lowest // batch) * batch
+        while floor <= highest:
+            yield (floor, floor + batch)
+            floor += batch
+
+    BOUNDARY_FINDER = """
+        SELECT MIN(transaction_id) AS floor, MAX(transaction_id) AS ceiling
+        FROM   {}"""
+
+    # Tuples of ( description, controlling table name, query)
+    UPDATERS = (('Place of performance for FABS', 'transaction_fabs', """
+            UPDATE references_location l
+            SET    state_name = UPPER(tf.place_of_perform_state_nam),
+                state_code = sa.abbrev
+            FROM   transaction_normalized tn
+            JOIN   transaction_fabs tf ON (tf.transaction_id = tn.id)
+            JOIN   state_abbrevs sa ON (UPPER(tf.place_of_perform_state_nam) = sa.name)
+            WHERE  tn.place_of_performance_id = l.location_id
+            AND    l.location_country_code = 'USA'
+            AND    l.state_code IS NULL
+            AND    l.state_name IS NULL
+            AND    l.place_of_performance_flag
+            AND    tf.transaction_id >= {floor}
+            AND    tf.transaction_id < {ceiling};
+            """), ('Recipient for FABS (get state code from name)', 'transaction_fabs', """
+        UPDATE references_location l
+        SET    state_name = UPPER(tf.legal_entity_state_name),
+               state_code = sa.abbrev
+        FROM   transaction_normalized tn
+        JOIN   transaction_fabs tf ON (tf.transaction_id = tn.id)
+        JOIN   legal_entity le ON (tn.recipient_id = le.legal_entity_id)
+        JOIN   state_abbrevs sa ON (UPPER(tf.place_of_perform_state_nam) = sa.name)
+        WHERE  le.location_id = l.location_id
+        AND    tf.legal_entity_state_code IS NULL
+        AND    l.location_country_code = 'USA'
+        AND    l.state_code IS NULL
+        AND    l.state_name IS NULL
+        AND    l.recipient_flag
+        AND    tf.transaction_id >= {floor}
+        AND    tf.transaction_id < {ceiling};
+        """), ('Recipient for FABS (get state name from code)', 'transaction_fabs', """
+        UPDATE references_location l
+        SET    state_name = sa.name,
+               state_code = UPPER(REPLACE(tf.legal_entity_state_code, '.', ''))
+        FROM   transaction_normalized tn
+        JOIN   transaction_fabs tf ON (tf.transaction_id = tn.id)
+        JOIN   legal_entity le ON (tn.recipient_id = le.legal_entity_id)
+        JOIN   state_abbrevs sa ON (UPPER(REPLACE(tf.legal_entity_state_code, '.', '')) = sa.abbrev)
+        WHERE  le.location_id = l.location_id
+        AND    tf.legal_entity_state_code IS NOT NULL
```

Comment:
Would this overlap any records that were recently filled in with their state codes if it's checking for `NOT NULL` here?

Reply:
Lines 90-91 mean that it shouldn't touch and location records that presently have a state code.